package.json

```
@@ -10,7 +10,11 @@
     "build": "tsc",
     "semantic-release": "semantic-release",
     "test": "jest src --coverage",
-    "test:e2e": "jest e2e",
+    "test:e2e": "jest e2e --runInBand",
+    "test:e2e:prep-crds": "kubectl apply -f test/ && npx ts-node src/cli.ts crd ./test/datastore.crd.yaml e2e && npx ts-node src/cli.ts crd https://raw.githubusercontent.com/defenseunicorns/kubernetes-fluent-client/refs/heads/471/test/webapp.crd.yaml e2e",
+    "test:e2e:prep-cluster": "k3d cluster create kfc-dev --k3s-arg '--debug@server:0' --wait && kubectl rollout status deployment -n kube-system",
+    "test:e2e:prep-image": "npm pack && npm i kubernetes-fluent-client-0.0.0-development.tgz --no-save",
+    "test:e2e:run": "npm run test:e2e:prep-cluster && npm run test:e2e:prep-crds && npm run test:e2e:prep-image && npm run test:e2e",
```

We also need to tear down the cluster after the tests run. Otherwise consecutive runs (like in local dev) will not work.
```suggestion
    "test:e2e:run": "npm run test:e2e:prep-cluster && npm run test:e2e:prep-crds && npm run test:e2e:prep-image && npm run test:e2e && npm run test:e2e:cleanup",
    "test:e2e:cleanup": "k3d cluster delete kfc-dev",
```

---

It should clean up on the afterAll but not if it fails but this is good for the case in which it does

```ts
  afterAll(async () => {
    try {
      execCommand(`k3d cluster delete ${clusterName}`);
    } catch {}
  });
```